### PR TITLE
Update fluentd-es-configmap.yaml

### PR DESCRIPTION
**Pull request by like-inspur**
set persistenet to false and add autosave interval config

from https://github.com/kubernetes/kubernetes/pull/94174, @coffeepac

### CODE REVIEW

**Comment by coffeepac** [comment]
@like-inspur can you also bump the following versions:
https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/fluentd-elasticsearch/fluentd-es-configmap.yaml#L4
https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/fluentd-elasticsearch/fluentd-es-ds.yaml#L114

and then for the daemonset: https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/fluentd-elasticsearch/fluentd-es-ds.yaml#L48-L64 but not the image version, no need for a new image for this.

**Comment by like-inspur** [comment]
@coffeepac have modified configmap version

**Comment by coffeepac** [comment]
@like-inspur apologies, I was not specific.

Configmap version should be set to v0.3.0, and the version of the daemonset should go from v3.0.5 -> v3.0.6.  We should _not_ modify the version of the image.

File: cluster/addons/fluentd-elasticsearch/fluentd-es-ds.yaml
Comment (coffeepac) [inline cluster/addons/fluentd-elasticsearch/fluentd-es-ds.yaml:114]: same as above, should be `v0.3.0`

**Review by coffeepac — COMMENTED** [review]
also please update these versions to `v3.0.6`. EDIT:  I had the version wrong here.  it was v3.1.0 but we don't need a minor bump, just a patch bump.
https://github.com/kubernetes/kubernetes/pull/95435/files#diff-5f26b4eba0e566463bf8fa83e05b2a7787115408e33b176c2d52e0d030a6a6b0R48
https://github.com/kubernetes/kubernetes/pull/95435/files#diff-5f26b4eba0e566463bf8fa83e05b2a7787115408e33b176c2d52e0d030a6a6b0R52
https://github.com/kubernetes/kubernetes/pull/95435/files#diff-5f26b4eba0e566463bf8fa83e05b2a7787115408e33b176c2d52e0d030a6a6b0R58
https://github.com/kubernetes/kubernetes/pull/95435/files#diff-5f26b4eba0e566463bf8fa83e05b2a7787115408e33b176c2d52e0d030a6a6b0R63

Thank you

**Comment by like-inspur** [comment]
@coffeepac have modified configmap version again

**Comment by coffeepac** [comment]
@like-inspur thank you for the configmap update.  

we also need to bump the version of the fluentd daemonset from v3.0.5 -> v3.0.6.  Its important that when someone pulls down v3.0.5 of the fluentd daemonset that the configuration doesn't change.  So, we need to bump the version.

**Comment by erain** [comment]
It makes sense to me to turn off persistent since the logs will go to ES anyway. 

@coffeepac : could you explain more why we need to bump the [fluentd-es daemonset version](https://github.com/kubernetes/kubernetes/blob/2729b8e375143434fc4977fe49eaea572567dac3/cluster/addons/fluentd-elasticsearch/fluentd-es-ds.yaml#L48). I understand your concern around configuration change, however the current daemonset version simply map to the fluentd image tag used, not a semver. So I am actually OK with the current change.

**Review by erain — APPROVED** [review]

**Comment by coffeepac** [comment]
@erain if we don't bump the version for the daemonset as a whole, then a user who does not alter their version but has to re-pull (cluster exploded, who knows) will get a _different and unexpected_ configmap and will have different functionality.  

I think this is a good change but we should also bump the patch version so that we do not surprise users with functionality changes.

**Comment by erain** [comment]
Looking at https://quay.io/repository/fluentd_elasticsearch/fluentd?tab=tags and there is no **image** at 3.1.1 version. How do we reconcile the image version and the daemonset version going forward?

**Comment by coffeepac** [comment]
@erain we don't.  I do not feel the need to keep the version of the daemonset and the version of the image the same.  they mean and refer to different things.

**Comment by erain** [comment]
My only concern is currently they do (otherwise what does `fluentd-es-v3.1.0` mean as the daemonset name?)

Personally I don't have strong opinion on that but I feel if we want to bump the daemonset version, we may want to choose a new version decoupled with image tag instead of bumping the current one. There will be likely a `v3.1.1` image version someday...

I will leave it to you and @dashpole to decide.

**Comment by coffeepac** [comment]
this is probably the right thing to do, I will try and take this over and get approval from probably @monotek

File: cluster/addons/fluentd-elasticsearch/fluentd-es-configmap.yaml
Comment (monotek) [inline cluster/addons/fluentd-elasticsearch/fluentd-es-configmap.yaml:346]: why do you remove persistence? 
its there to be able to pickup the buffer after a container crash.
otherwise you might loose some logs.

File: cluster/addons/fluentd-elasticsearch/fluentd-es-configmap.yaml
Comment (monotek) [inline cluster/addons/fluentd-elasticsearch/fluentd-es-configmap.yaml:361]: why do you remove persistence?
its there to be able to pickup the buffer after a container crash.
otherwise you might loose some logs.

File: cluster/addons/fluentd-elasticsearch/fluentd-es-configmap.yaml
Comment (monotek) [inline cluster/addons/fluentd-elasticsearch/fluentd-es-configmap.yaml:375]: why do you remove persistence?
its there to be able to pickup the buffer after a container crash.
otherwise you might loose some logs.

File: cluster/addons/fluentd-elasticsearch/fluentd-es-configmap.yaml
Comment (monotek) [inline cluster/addons/fluentd-elasticsearch/fluentd-es-configmap.yaml:389]: why do you remove persistence?
its there to be able to pickup the buffer after a container crash.
otherwise you might loose some logs.

File: cluster/addons/fluentd-elasticsearch/fluentd-es-configmap.yaml
Comment (monotek) [inline cluster/addons/fluentd-elasticsearch/fluentd-es-configmap.yaml:403]: why do you remove persistence?
its there to be able to pickup the buffer after a container crash.
otherwise you might loose some logs.

**Review by monotek — CHANGES_REQUESTED** [review]

File: cluster/addons/fluentd-elasticsearch/fluentd-es-ds.yaml
Comment (like-inspur) [reply cluster/addons/fluentd-elasticsearch/fluentd-es-ds.yaml:114]: ok

File: cluster/addons/fluentd-elasticsearch/fluentd-es-configmap.yaml
Comment (like-inspur) [reply cluster/addons/fluentd-elasticsearch/fluentd-es-configmap.yaml:346]: I don't found lose some logs, but when config persitent true, disk io on node will be very high

File: cluster/addons/fluentd-elasticsearch/fluentd-es-configmap.yaml
Comment (coffeepac) [reply cluster/addons/fluentd-elasticsearch/fluentd-es-configmap.yaml:347]: @monotek I asked for this in the original PR in order to make this particular change obvious.

**Comment by coffeepac** [comment]
the PR I've opened is this branch with two extra edits:
- rev version of daemonset
- rebase against master

@like-inspur if you want to port those changes to here I'll approve here

And apologies for the slow response on my part.

**Comment by like-inspur** [comment]
@coffeepac you can merge directly on Like inspur/master #100168

**Comment by coffeepac** [comment]
@like-inspur I tried that last night.  Have you granted me permissions since then?  If so, I will try again today at some point and close my PR.

**Comment by monotek** [comment]
I'm still not sure if persistence should be disabled by default.
Maybe people will more care for complete logs as for disk io but i guess it depends on the environment.

**Comment by coffeepac** [comment]
@like-inspur I have attempted to push to your fork and I am getting permission denied errors.  If you want to either a. grant me privs or b. grab the commit off my branch that's fine.

**Comment by coffeepac** [comment]
@monotek I believe based on a comment from like-inspur in the prior PR (https://github.com/kubernetes/kubernetes/pull/94174#issuecomment-694609298) this shouldn't result in dropped logs.  It might result in duplicated messages if fluentd reads from file, forwards the logs and then crashes before updating the position file.

**Pull request closed without merging**
